### PR TITLE
monitoring: Clean up Grafana dashboards

### PR DESCRIPTION
**Pull request by p-se**
Fixes few smaller issues. The issues are mentioned in broader detail in the tracker ticket or the corresponding commit messages.

Fixes: https://tracker.ceph.com/issues/51212

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by p-se** [comment]
As I was thinking about and picking suggestions from #41047, I realized that the commit e58c226, which makes a widget of the Ceph Cluster dashboard also filterable by the `instance` label might not be necessary. 

Usually instance labels are created by Prometheus when exporters are being scraped. Prometheus puts the `instance` label on a metric and sets the value to the instance it scraped the data from. If the exporter happens to export data related to the host, that's great. If it doesn't, the instance label doesn't has much meaning.

In case of the Prometheus MGR module (Ceph exporter), data from the Ceph cluster is provided on a single node (mgr node) for the whole cluster. This differs much from how, for instance, node-exporter works. If the Prometheus MGR module doesn't add a meaningful `instance` label itself and that's one used, there's not much use to the instance label added by Prometheus.

Hence, I'm thinking about removing the instance label from the Ceph Cluster dashboard altogether. And maybe even from some of the other dashboards, if it makes sense. This would not (and shouldn't) affect any data exported by the node-exporter.

For the Ceph Cluster dashboard it might even make most sense to do so. I forced a mgr daemon to fail, so that another took over and continued serving metrics. The Ceph Cluster dashboard is only complete if it isn't filtered by any instance, as only one mgr instance at a time can run mgr modules. So, only if all metrics are taken into account, the view of that dashboard will be most complete.

I'm not sure though, if I missed anything. Can you think of something why the instance label, at least for the Ceph Cluster dashboard, couldn't be removed?

File: monitoring/grafana/dashboards/ceph-cluster.json [hunk 1]
```diff
@@ -1245,7 +1245,7 @@
       "30d"
     ]
   },
-  "timezone": "browser",
+  "timezone": "",
```
Comment (epuertat) [inline monitoring/grafana/dashboards/ceph-cluster.json:1248]: In order to enforce this, I'd suggest that you add an assertion in the `ci/check_grafana_uids.py` script (and probably we can rename it that to `check_grafana_dashboards` or something) to ensure this cannot be (unintentionally) changed.

File: monitoring/grafana/dashboards/rbd-details.json [hunk 2]
```diff
@@ -1,409 +1,409 @@
...
\ No newline at end of file
+{
+  "__inputs": [],
+  "__requires": [
+    {
+      "type": "grafana",
+      "id": "grafana",
+      "name": "Grafana",
+      "version": "5.3.3"
+    },
```
Comment (epuertat) [inline monitoring/grafana/dashboards/rbd-details.json:9]: Don't get it... I diff'ed these 2 files and are exactly the same. So why are they here (some attribute) 🤔 ?

**Review by epuertat — APPROVED** [review]
LGTM, but I recommend adding a 'unit test' to avoid the timezone issue from happening again.

File: monitoring/grafana/dashboards/rbd-details.json (hunk 2 above)
Comment (p-se) [reply monitoring/grafana/dashboards/rbd-details.json:9]: I've converted the line endings from CRLF to LF. It's explained in the commit message.

**Comment by pcuzner** [comment]
@p-se IIRC 'we' used to apply instance on the mgr job to provide a generic name, so no matter where the metrics were scraped from the instance name was consistent. 

Having said that I think that given we have a 1:1 relationship ceph:prometheus the instance issue is not going to be a problem.

**Comment by p-se** [comment]
> @p-se IIRC 'we' used to apply instance on the mgr job to provide a generic name, so no matter where the metrics were scraped from the instance name was consistent.
> 
> Having said that I think that given we have a 1:1 relationship ceph:prometheus the instance issue is not going to be a problem.

Thanks for sharing! I also don't think that having the ability to filter by instance is a real problem, it's just that it doesn't seem to make any difference. And if that's the case, it's probably just confusing to offer the ability to filter dashboards/panels by instance without a functionality behind it. In addition, we have the "filtering" in the PromQL queries as if it would make a difference/sense to have them there. That why I'm wondering if it shouldn't be removed completely from all those places where it doesn't make sense, to neither confuse contributors nor users. This is most evident on the "Ceph - Cluster" Grafana dashboards, which is, by the way, not displayed by Ceph Dashboard. But I think there might be some places in other dashboards where removing the instance label from PromQL queries would probably make sense.

**Comment by p-se** [comment]
I had the feeling that I might not have been able to explain what it's about with that `instance` label. That I didn't implement it but kept the first idea of the implementation probably hadn't helped. So I implemented it now.  Please have a look at https://github.com/ceph/ceph/pull/41838/commits/037410713f032c0a2a25243e411ae67dffcc1d1a.

This is what it looks like now, with all the other commits/fixes:

removed_instance_label

Note that the option to filter by instance is gone. It previously would only have shown one instance to filter from, assuming the mgr instance hasn't failed recently. Only if the mgr instance had failed, it would offer two instances to chose from. By deciding for either one, data would be split between what has been collected from the one or the other instance.

The missing data of the OSDs, Monitors and Client Connections panels is an [issue with the status panel plugin](https://tracker.ceph.com/issues/45908) I'm already working on.

**Comment by alfonsomthd** [comment]
> I had the feeling that I might not have been able to explain what it's about with that `instance` label. [...]

@p-se Thanks for the clarification.

**Comment by jan--f** [comment]
Re. the instance discussion: iirc the `instance` template variable was originally added to be able to monitor multiple clusters. That however requires some special config for the reasons mentioned further up. There are two approaches that are detailed in the [docs](https://docs.ceph.com/en/latest/mgr/prometheus/#correlating-drive-statistics-with-node-exporter). One has to either rewrite the instance label in the query or add the `honor_labels` config to get a consistent `instance` label across all mgr instances.

**Review by avanthakkar — APPROVED** [review]

File: src/pybind/mgr/dashboard/ci/check_grafana_dashboards.py [hunk 3]
```diff
@@ -79,6 +79,10 @@ def get_grafana_dashboards(base_dir):
                 assert dashboard_config['id'] is None, \
                     "'id' not null: '{}'".format(dashboard_config['id'])
 
+                assert 'timezone' not in dashboard_config or dashboard_config['timezone'] == '', \
+                    ("'timezone' field must not be set to anything but an empty string or be "
+                     "omitted completely")
+
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/ci/check_grafana_dashboards.py:85]: 👏🏼 thanks!

**Review by epuertat — APPROVED** [review]

**Pull request merged**